jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/conf/JdbcStorageConfigManager.java

```diff
@@ -77,6 +81,15 @@ public static void copySecretsToJob(Properties props, Map<String, String> jobSec
       String key = props.getProperty(CONFIG_PWD_KEY);
       passwd = Utilities.getPasswdFromKeystore(keystore, key);
     }
+    if (passwd == null) {
```

Comment:
What is the order of precedence here? if user specifies password in properties, also specifies a keystore with password and password store URI, looks like we will pick it up from the properties which is the least secure means of specifying it. Should we reverse the order and look for password store first? just thinking out loud.
or should we prevent the user from specifying multiple means.

Reply:
Make sense, I'll add a check to error out if multiple password sources are set only one of password, keystore or uri should be set. And will do the same in JdbcDatabaseAccessor. I was a bit lazy, I think I'll refactor the code to get any password.